dallinger/prolific.py

```diff
@@ -281,3 +282,84 @@ def _req(self, method: str, endpoint: str, **kw) -> dict:
             raise ProlificServiceException(json.dumps(error))
 
         return parsed
+
+
+class DevProlificService(ProlificService):
+    """Wrapper that mocks the Prolific REST API and instead of making requests it writes to the log."""
+
+    def _req(self, method: str, endpoint: str, **kw) -> dict:
+        """Does NOT make any requests but instead writes to the log."""
+        self.debug_log(f"method = {method}, endpoint = {endpoint}, kw = {kw}")
+
+        if endpoint.startswith("/studies/"):
+            if method == "GET":
+                # method="GET", endpoint=f"/studies/{study_id}/"
+                if re.match(r"/studies/[a-z0-9]+/", endpoint):
+                    return {
+                        "id": "prolific-user-id",
+                        "external_study_url": "external-study-url",
+                    }
+                # method="GET", endpoint="/studies/"
+                elif endpoint == "/studies/":
+                    return {
+                        "results": [
+                            {
+                                "title": "title",
+                                "status": "status",
+                                "created": "created-date",
+                                "expiration": "",
+                                "description": "",
+                            }
+                        ]
+                    }
+
+            elif method == "POST":
+                # method="POST", endpoint="/studies/", json=payload
+                if endpoint == "/studies/":
+                    return {
+                        "id": "study-id",
+                        "external_study_url": "external-study-url",
+                    }
+
+                # method="POST", endpoint=f"/studies/{study_id}/transition/", json={"action": "PUBLISH"},
+                elif re.match(r"/studies/[a-z0-9]+/transition/", endpoint):
+                    return True
+
+            elif method == "PATCH":
+                # method="PATCH", endpoint=f"/studies/{study_id}/", json={"total_available_places": new_total},
+                return {
+                    "items": ["https://experiment-url-1", "https://experiment-url-2"],
+                    "message": "More info about this particular recruiter's process",
+                }
+
+            # method="DELETE", endpoint=f"/studies/{study_id}"
+            elif method == "DELETE":
+                return {"status_code": 204}
+
+        # method="POST", endpoint=f"/bulk-bonus-payments/{setup_response['id']}/pay/"
+        elif endpoint.startswith("/bulk-bonus-payments/"):
+            return {"id": "id-from call-to-/bulk-bonus-payments/<id>/pay/"}
```

Comment:
Returning IDs in this format creates strange looking (invalid) API calls down the line, e.g.

```
endpoint = /bulk-bonus-payments/id-from call-to-/submissions/bonus-payments/pay/, kw = {}
```

Can I suggest instead that we generate a random ID?

```py
from uuid import uuid4

str(uuid4())
```